src/script-events.c

```diff
@@ -0,0 +1,337 @@
+        if (br == '/')                                                      \
+            goto Skip;                                                      \
+        br = 0;                                                             \
+    }
+
+#define SEND_PATH_N(N) REPEAT_##N(SEND_PATH;)
+
+#define READ_CHAR_STR                                                   \
+    ev.id = id;                                                         \
+    ev.done = FALSE;                                                    \
+    ev.telemetry_type = TE_CHAR_STR;                                    \
+    __builtin_memset(&ev.u.v.value, 0, VALUE_SIZE);                     \
+    count = bpf_probe_read_str(&ev.u.v.value, VALUE_SIZE, (void *)str); \
+    if (count >= VALUE_SIZE)                                            \
+    {                                                                   \
+        ev.u.v.truncated = TRUE;                                        \
+        str = str + VALUE_SIZE - 1;                                     \
+        push_telemetry_event(ctx, &ev);                                 \
+    }                                                                   \
+    else                                                                \
+    {                                                                   \
+        if (count > 0)                                                  \
+        {                                                               \
+            ev.u.v.truncated = FALSE;                                   \
+            push_telemetry_event(ctx, &ev);                             \
+            goto DONE_READING;                                          \
+        }                                                               \
+    }
+
+#define READ_CHAR_STR_N(N) REPEAT_##N(READ_CHAR_STR)
+
+static __always_inline void push_telemetry_event(struct pt_regs *ctx, ptelemetry_event_t ev)
+{
+    bpf_perf_event_output(ctx, &script_events, bpf_get_smp_processor_id(), ev, sizeof(*ev));
+    __builtin_memset(ev, 0, sizeof(telemetry_event_t));
+}
+
+SEC("kprobe/script_load")
+int kprobe__script_load(struct pt_regs *ctx)
+{
+    struct linux_binprm *bprm = (struct linux_binprm *)PT_REGS_PARM1(ctx);
+    u32 index = (u32)bpf_get_current_pid_tgid();
+
+    bpf_map_update_elem(&load_script_map, &index, &bprm, BPF_ANY);
+
+    return 0;
+}
+
+SEC("kprobe/handle_pwd")
+int kprobe__handle_pwd(struct pt_regs *ctx)
+{
+    int count = 0;
+    char br = 0;
+    telemetry_event_t sev = {0};
+    ptelemetry_event_t ev = &sev;
+    unsigned long long ret = 0;
+
+    // if the ID already exists, we are tail-calling into ourselves, skip ahead to reading the path
+    u64 p_t = bpf_get_current_pid_tgid();
+    u64 id = (u64)bpf_map_lookup_elem(&process_ids, &p_t);
+    if (id)
+    {
+        __builtin_memcpy(&id, (void *)id, sizeof(u64));
+    }
+
+    // the verifier complains if these are instantiated before the Pwd:; label
```

Comment:
```suggestion
```

Reply:
Good catch. Cleaning up comments is important